src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -288,6 +288,7 @@ public override Binder VisitAccessorDeclaration(AccessorDeclarationSyntax parent
                                     if ((object)propertySymbol != null)
                                     {
                                         accessor = (parent.Kind() == SyntaxKind.GetAccessorDeclaration) ? propertySymbol.GetMethod : propertySymbol.SetMethod;
+                                        Debug.Assert(accessor is not null || parent.HasErrors);
```

Comment:
Is there any way (maybe through speculation) to get a binder inside a property without any accessor?
Then there would be no accessor, but the parent syntax also would have no error (we report the lack of accessor as a binding error)

Reply:
I'm going to add a prototype comment to make sure we close on this. For now, I am going to leave the assert as-is, because it has been helpful in ensuring we are getting the expected accessor symbol in partial property scenarios, and no test case exists yet which is failing the assertion.

When we do come up with such scenario that fails the assertion, we could further refine the assert, or potentially just remove it, because at that time the necessary adjustments to GetPropertySymbol have already been made and tested.

Reply:
I'm suggesting that we tighten the assertion to `Debug.Assert(accessor is not null);`.
If we can come up with a scenario where that's violated, then we'll hit it and relax the assertion. No need for a PROTOTYPE comment.

Reply:
I used that condition in the assertion earlier on and it failed in some tests, inserting `|| parent.HasErrors` resolved the issue.